### ISSUE TITLE
Broken urls in the xml output

### ISSUE THREAD

**Issue by skypjack** [doxygen/doxygen#7477]
Steps to reproduce the problem:

* Create an `.md` file with at least the following line:
   ```
   ![Alt](https://foobar.com/quux/img.png)
   ```
   This is how we include images eg on the README file of a project on GH.

* Configure doxygen so as to generate the xml output and to use the md file for the main page:
   ```
   USE_MDFILE_AS_MAINPAGE = my_file.md
   GENERATE_XML = YES
   ```

* Generate the xml output. This is what we obtain:
   ```
   <image type="html" name="img.png" inline="yes">
   ```
   Long story short, doxygen strips out the url for unknown reasons and the resulting path is literally unusable.

Note that the html generator returns it correctly instead:
```
<img src="https://foobar.com/quux/img.png" alt="Alt" class="inline"/>
```

I'm not sure which of the two behaviors is the correct one even if I would favor the second one.
However, it would be useful if the two generators were consistent when it comes to dealing with urls.

Tested with doxygen v1.8.16 - no idea if the latest version suffers from the same problem but I guess so since I can't find any issue for that nor something similar in the changelog.

CC @mosra

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7479 

Note: in future can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent. This will make testing a bit easier (download file and unpack).

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by skypjack** [comment]
@albert-github ok, next time I'll try to submit a proper example. Thank you for looking into it anyway. :+1:

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.18.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
